### PR TITLE
accept arbitrary path length for file

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**
changes SSL_CTX_load_verify_locations to accept path longer than 256 characters.

### CODE REVIEW

File: src/ssl.cpp [hunk 1]
```diff
@@ -792,48 +792,68 @@ int SSL_CTX_load_verify_locations(SSL_CTX* ctx, const char* file,
         WIN32_FIND_DATA FindFileData;
         HANDLE hFind;
 
-        char name[MAX_PATH + 1];  // directory specification
-        strncpy(name, path, MAX_PATH - 3);
+        char* name = NEW_YS char[strlen(path) + 4];  // directory specification
```
Comment (toddouska) [inline src/ssl.cpp]: strlen(path) is used several times, I think it should be cached.

File: src/ssl.cpp [hunk 2]
```diff
@@ -792,48 +792,68 @@ int SSL_CTX_load_verify_locations(SSL_CTX* ctx, const char* file,
...
+        int nameSz = strlen(path) + 4; // 4 for extra space needed with string
+        strncpy(name, path, nameSz - 3);
```
Comment (toddouska) [inline src/ssl.cpp]: Magic numbers should be removed.   These should have names.  DIR_SEP_SZ, DIR_SEP_STAR_SZ?  +1 can denote terminator but the comment should reflect that.

File: src/ssl.cpp [hunk 3]
```diff
@@ -792,48 +792,68 @@ int SSL_CTX_load_verify_locations(SSL_CTX* ctx, const char* file,
...
-                strncpy(name, path, MAX_PATH - 2 - HALF_PATH);
+            if (!(FindFileData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)) {
+                if ((int)strlen(path) + 4 +
+                                 (int)strlen(FindFileData.cFileName) > nameSz) {
+                    ysArrayDelete(name);
+                    nameSz = strlen(path) + 4 + strlen(FindFileData.cFileName);
+                    name = NEW_YS char[nameSz];
+                }
+                memset(name, 0, nameSz);
+                strncpy(name, path, nameSz);
```
Comment (toddouska) [inline src/ssl.cpp]: While path len is less than nameSz, a strncpy of full buffer size could leave an unterminated string which is a bug.

File: src/ssl.cpp [hunk 4]
```diff
@@ -792,48 +792,68 @@ int SSL_CTX_load_verify_locations(SSL_CTX* ctx, const char* file,
...
                 strncat(name, "\\", 2);
-                strncat(name, FindFileData.cFileName, HALF_PATH);
+                strncat(name, FindFileData.cFileName,
+                                                     nameSz - strlen(path) - 1);
```
Comment (toddouska) [inline src/ssl.cpp]: path len and 2 chars have been added.   -1 is bug.  Even -2 is a bug because it could leave an unterminated string.

**Comment by JacobBarthelmeh** [comment]
Thanks for reviewing the code Todd. Will work on the updating the PR for the changes.

**Comment by toddouska** [comment]
Thanks for the changes, looks good.

**Pull request merged**